packages/gatsby-recipes/src/renderer/render.test.js

```diff
@@ -10,6 +12,38 @@ const fixture = (
   </doc>
 )
 
+test(`handles nested rendering`, async () => {
+  const result = await render(
+    <doc>
+      <File path="red.js" content="red!">
+        <File path="blue.js" content="blue!">
+          <File path="yellow.js" content="yellow!" />
+        </File>
+      </File>
+    </doc>,
+    {}
+  )
+
+  expect(result).toMatchInlineSnapshot(`
+    Array [
+      Object {
+        "currentState": "",
+        "describe": "Write red.js",
+        "diff": "- Original  - 0
+    + Modified  + 1
+
+    + red!",
+        "newState": "red!",
```

Comment:
does this snapshot need updated? It doesn't have the blue/yellow files in it

Reply:
The transformer doesn't currently handle nested children, I figured I'd follow along with that as a separate PR into this one (I'll PR that functionality into this branch).

Reply:
Resolved with: https://github.com/johno/gatsby/pull/2